### PR TITLE
Added unit test cases for Preflight Controller

### LINKED ISSUE #399: Add unit tests for FileUpload Vulnerability
**Is your feature request related to a problem? Please describe.**
We are looking at increasing the Test coverage for the project for its stability so this task is to add junit tests for https://github.com/SasanLabs/VulnerableApp/tree/master/src/main/java/org/sasanlabs/service/vulnerability/fileupload

### PR DESCRIPTION
**Pull request by shlokam**
Attempts to fix #399

### CODE REVIEW

**Comment by shlokam** [comment]
Can you please let me know why the pipeline failed?

File: src/main/java/org/sasanlabs/internal/utility/FileReader.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.sasanlabs.service.vulnerability.fileupload.UnrestrictedFileUpload;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
+
+public class FileReader {
```
Comment (preetkaran20) [inline src/main/java/org/sasanlabs/internal/utility/FileReader.java:11]: I think you have created this class such that you can mock the `readAsStream` functionality right?
I think you don't need a different class, create a `spy` for  PreflightController and a method in PreflightController which you can mock using spy and everything else will work perfectly.

Comment (shlokam) [reply src/main/java/org/sasanlabs/internal/utility/FileReader.java:11]: Yes, that was the purpose for the class. 
Ok. I will make the changes.

File: src/test/java/org/sasanlabs/service/vulnerability/fileUpload/PreflightControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.sasanlabs.service.vulnerability.fileupload.UnrestrictedFileUpload.CONTENT_DISPOSITION_STATIC_FILE_LOCATION;
+
+public class PreflightControllerTest {
+
+    @Mock
+    UnrestrictedFileUpload unrestrictedFileUpload;
```
[the remark this answers is not in the document]
Comment (shlokam) [reply src/test/java/org/sasanlabs/service/vulnerability/fileUpload/PreflightControllerTest.java:31]: Ok

**Pull request closed without merging**